stock_reserve_sale_mrp/wizard/sale_stock_reserve.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class SaleStockReserve(models.TransientModel):
+    _inherit = "sale.stock.reserve"
+
+    def _prepare_stock_reservation_bom_line(self, line, bom_line, bom_vals):
+        self.ensure_one()
+        return {
+            "product_id": bom_line.product_id.id,
+            "product_uom": bom_line.bom_id.product_uom_id.id,
+            "product_uom_qty": bom_vals["qty"],
+            "date_validity": self.date_validity,
+            "name": "{} ({})".format(line.order_id.name, line.name),
+            "location_id": self.location_id.id,
+            "location_dest_id": self.location_dest_id.id,
+            "note": self.note,
+            "sale_line_id": line.id,
+            "restrict_partner_id": self.owner_id.id,
+        }
+
+    def stock_reserve(self, line_ids):
+        self.ensure_one()
+        lines = self.env["sale.order.line"].browse(line_ids)
+        lines_without_kit = self.env["sale.order.line"]
+        vals_list = []
+        for line in lines:
+            bom_kit = self.env["mrp.bom"]._bom_find(
+                product=line.product_id,
+                company_id=line.company_id.id,
+                bom_type="phantom",
+            )
```

Comment:
Is it possible to use the BOM from the stock move? just wondering

```suggestion
            bom_kit =  line.move_ids.filtered(lambda m: m.state != 'cancel').mapped('bom_line_id.bom_id').filtered(lambda b: b.type == 'phantom')
```

Reply:
Hi @AaronHForgeFlow, at this point the moves are not created. The aim of this module and `stock_reserve_sale` is to allow reservations on the quotation state of the SO.

Reply:
ok! thanks for letting me know!